feat: implement view transition functionality

jairusjoer opened the pull request:
## Description

This pull request implements [View Transitions API](https://developer.mozilla.org/en-US/docs/Web/API/View_Transition_API) support for smoother page navigation in Quartz's SPA functionality.

## Changes

- **Added `startViewTransition` utility function** in `quartz/components/scripts/util.ts` that wraps code execution with the View Transitions API when available
- **Updated SPA navigation** in `quartz/components/scripts/spa.inline.ts` to use view transitions during page morphing
- **Added view transition styles** in `quartz/styles/base.scss` to disable pointer events during transitions

## Details

The implementation wraps the existing `micromorph` document body update and related DOM operations within a `startViewTransition` call. This creates smooth, animated transitions between pages when the View Transitions API is supported.

For browsers without View Transitions API support, the code gracefully falls back to the original behavior, ensuring backward compatibility.

## Rationale

- Uses browser-native features, no additional dependencies
- Provides progressive enhancement for baseline browsers
- Maintains previous behavior with older browsers
- Preserves the original functionality completely

## System

- **Quartz**: v4.5.2
- **Node**: v22.18.0
- **npm**: v10.9.3
- **OS**: macOS
- **Browser**: Dia (Chromium)

jairusjoer committed:
feat: implement view transition functionality

aarnphm reviewed (approved):
I'm fine with this, maybe cc @jackyzha0 and @saberzero1 on this as well. not really a big fan of animation, but this will break the progress bar logic?

jairusjoer commented:
@aarnphm I didn't encounter any issues with the progress bar when I tested it using network throttling. If there's something I've missed, please let me know 💪

saberzero1 commented:
> I'm fine with this, maybe cc @jackyzha0 and @saberzero1 on this as well. not really a big fan of animation, but this will break the progress bar logic?

I'm not a fan of the graph flashing white on page transition.

Perhaps the transition should be configurable? Both whether it is enabled/disabled and the exact transition.

jairusjoer committed:
feat: add configuration option for enabling browser-native view transitions

jairusjoer committed:
feat: add doucmentation for view transitions

jairusjoer commented:
I've addressed the mentioned issues and concerns:

- White flashing of graph has been eliminated
- View transitions are now configurable via the `enableViewTransitions` key
- Tagged key components with view transition names to allow customization
- Added Information and further resources on customization of transitions

Let me know what you think! ✌️

aarnphm commented on docs/features/View Transitions.md:
````diff
@@ -0,0 +1,63 @@
+})
+```
+
+The `startViewTransition` function will automatically use the View Transitions API if it's supported and enabled, or fall back to immediate execution otherwise.
````
```suggestion
The `startViewTransition` function will automatically use the View Transitions API if the browser is supported, otherwise it will fallback to eager execution.
```

jairusjoer committed:
Update docs/features/View Transitions.md

Co-authored-by: Aaron Pham <contact@aarnphm.xyz>

jackyzha0 reviewed (approved):
we should think about backwards compat, startViewTransition unfortunately only stabilized in firefox on ⁨2025-10-14⁩

small weird style thing (looks like a stacking order problem?)

<img width="1608" height="937" alt="Image" src="https://github.com/user-attachments/assets/76d585c7-a02c-4a19-8c70-0c0e8b851bda" />

generally feels fine but i would default it to off before merging

jairusjoer committed:
fix: disable view transitions  for global graph and adjust graph styles

jairusjoer commented:
@jackyzha0 The provided implementation is backwards-compatible and has been set up according to an [MDN example](https://developer.mozilla.org/en-US/docs/Web/API/ViewTransition#examples). I tested the implementation on Firefox 130.0 (released on 3 September 2024) with both view transitions enabled and disabled, and found no issues with fallback behavior.

The stack ordering issue has been resolved and the feature is now disabled by default. As the issue occurred even with the feature disabled, this should not affect the review.

Additionally, I encountered a pre-existing issue while inspecting the global graph view, whereby the graph overflowed on smaller viewports. I have also addressed this:

<img width="625" height="592" alt="Screenshot 2025-10-28 at 10 02 42" src="https://github.com/user-attachments/assets/bdfe9538-7fe3-4c3e-a298-41d47e9c7085" />
